python/pyarrow/src_arrow/CMakeLists.txt

```diff
@@ -0,0 +1,467 @@
+
+  set(FLIGHT_PROTO_PATH "${ARROW_SOURCE}/format")
+  set(FLIGHT_PROTO "${ARROW_SOURCE}/format/Flight.proto")
+
+  set(FLIGHT_GENERATED_PROTO_FILES
+      "${ARROW_SOURCE_DIR}/build/src/arrow/flight/Flight.pb.cc"
+      "${ARROW_SOURCE_DIR}/build/src/arrow/flight/Flight.pb.h"
+      "${ARROW_SOURCE_DIR}/build/src/arrow/flight/Flight.grpc.pb.cc"
+      "${ARROW_SOURCE_DIR}/build/src/arrow/flight/Flight.grpc.pb.h")
+
+  set(PROTO_DEPENDS ${FLIGHT_PROTO} gRPC::grpc_cpp_plugin)
+
+  add_custom_command(OUTPUT ${FLIGHT_GENERATED_PROTO_FILES}
+                    COMMAND ${ARROW_PROTOBUF_PROTOC} "-I${FLIGHT_PROTO_PATH}"
+                            "--cpp_out=${${ARROW_SOURCE_DIR}/build/src/arrow/flight/}" "${FLIGHT_PROTO}"
+                    DEPENDS ${PROTO_DEPENDS} ARGS
+                    COMMAND ${ARROW_PROTOBUF_PROTOC} "-I${FLIGHT_PROTO_PATH}"
+                            "--grpc_out=${${ARROW_SOURCE_DIR}/build/src/arrow/flight/}"
+                            "--plugin=protoc-gen-grpc=$<TARGET_FILE:gRPC::grpc_cpp_plugin>"
+                            "${FLIGHT_PROTO}")
+
+  set_source_files_properties(${FLIGHT_GENERATED_PROTO_FILES} PROPERTIES GENERATED TRUE)
+  add_custom_target(flight_grpc_gen ALL DEPENDS ${FLIGHT_GENERATED_PROTO_FILES})
+
+  add_arrow_lib(arrow_python_flight
+                CMAKE_PACKAGE_NAME
+                ArrowPythonFlight
+                PKG_CONFIG_NAME
+                arrow-python-flight
+                SOURCES
+                flight.cc
+                OUTPUTS
+                ARROW_PYFLIGHT_LIBRARIES
+                DEPENDENCIES
+                flight_grpc_gen
+                SHARED_LINK_FLAGS
+                ${ARROW_VERSION_SCRIPT_FLAGS} # Defined in line 95
+                SHARED_LINK_LIBS
+                arrow_python_shared
+                arrow_flight_shared
+                STATIC_LINK_LIBS
+                ${PYTHON_OTHER_LIBS}
+                EXTRA_INCLUDES
+                "${ARROW_PYTHON_INCLUDES}"
+                PRIVATE_INCLUDES
+                "${Protobuf_INCLUDE_DIRS}")
+
+  add_dependencies(arrow_python ${ARROW_PYFLIGHT_LIBRARIES})
+
+  foreach(LIB_TARGET ${ARROW_PYFLIGHT_LIBRARIES})
+    target_compile_definitions(${LIB_TARGET} PRIVATE ARROW_PYFLIGHT_EXPORTING)
+  endforeach()
+
+  if(ARROW_BUILD_STATIC AND MSVC)
+    target_compile_definitions(arrow_python_flight_static PUBLIC ARROW_STATIC)
+  endif()
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  # Clang, be quiet. Python C API has lots of macros
+  set_property(SOURCE ${ARROW_PYTHON_SRCS}
+               APPEND_STRING
+               PROPERTY COMPILE_FLAGS -Wno-parentheses-equality)
+endif()
+
+arrow_install_all_headers("arrow/python")
+
+# ----------------------------------------------------------------------
+
+if(ARROW_BUILD_TESTS)
+  enable_testing()
+
+  # Set necessary paths for cmake to find GTest
+  set(GTEST_ROOT ${ARROW_SOURCE_DIR}/build/googletest_ep-prefix)
```

Comment:
We can't assume that all users use `${ARROW_SOURCE_DIR}/build` as a build directory.
(For example, I always use `cpp.build` as a build directory.)

How about accepting `ARROW_BUILD_DIR` CMake variable from users to detect build directory for Apache Arrow C++?

Reply:
Oh, haven't thought about that.
Sure, that would be easy to implement and I think it is a good idea.